src/stream/pipeline/pipeline.rs

```diff
@@ -129,48 +142,44 @@ impl PipelineState {
 
     /// Links the sink pad from the given Sink to this Pipeline's Tee element
     #[instrument(level = "debug")]
-    pub fn add_sink(&mut self, sink: Sink) -> Result<()> {
-        let sink_id = &sink.get_id().clone();
+    pub fn add_sink(&mut self, mut sink: Sink) -> Result<()> {
+        let pipeline_id = &self.pipeline_id;
 
         // Request a new src pad for the Tee
-        let tee_src_pad = self
-            .tee
-            .request_pad_simple("src_%u")
-            .context("Failed requesting src pad for Tee")?;
-        let sink = sink.set_tee_src_pad(tee_src_pad);
+        let tee_src_pad = self.tee.request_pad_simple("src_%u").context(format!(
+            "Failed requesting src pad for Tee of the pipeline {pipeline_id}"
+        ))?;
+        debug!("Got tee's src pad {:#?}", tee_src_pad.name());
 
-        // Add the Sink element to the Pipeline
         let pipeline = &self.pipeline;
-        let sink_element = sink.get_element();
-        pipeline.add(sink_element.as_ref() as &gstreamer::Element)?;
-
-        // Link the new Tee's src pad to the Sink's sink pad
-        let sink_pad = sink.get_sink_pad();
-        let tee_src_pad = sink.get_tee_src_pad().context("Tee src pad is None")?;
-        if let Err(error) = tee_src_pad.link(sink_pad) {
-            pipeline.remove(sink_element.as_ref() as &gstreamer::Element)?;
-            bail!(error)
-        }
 
-        pipeline.debug_to_dot_file(
-            gstreamer::DebugGraphDetails::all(),
-            format!("sink-{}-before-playing", &sink.get_id()),
+        // Temporarely set to Ready to avoid losing frames during connection of the new Sink
+        // if let Err(error) = pipeline.set_state(gst::State::Ready) {
+        //     sink.unlink(pipeline, &self.pipeline_id)?;
+        //     bail!(error)
+        // }
```

Comment:
I'm investigating a good way to do this. If I set the state to Ready, all already-playing webrtc sessions (the other webrtcbin sinks) stop and don't return after the `pipeline.set_state(gst::State::Playing)` that is being done below in this function. 

One idea to try is to ask it for syncronize its state with its pipeline state, or to set its state to playing again. But if the problem is because it couldn't be changed because of some internal behavior, then I'll try to temporarily block the set_state signal for it.

Reply:
Fixed!